drivers/ethernet/phy/Kconfig

```diff
@@ -41,13 +41,34 @@ config PHY_ADIN2111
 	  Enable ADIN2111 PHY driver.
 
 config PHY_MICROCHIP_KSZ8081
-	bool "Microchip KSZ8081 Phy Driver"
+	bool "Microchip KSZ8081 PHY Driver"
 	default y
 	depends on DT_HAS_MICROCHIP_KSZ8081_ENABLED
 	depends on MDIO
 	depends on GPIO
 	help
-	  Enable Microchip KSZ8081 Ethernet Phy Driver
+	  Enable Microchip KSZ8081 Ethernet PHY Driver
+
+config PHY_REALTEK_RTL8211F
+	bool "Realtek RTL8211F PHY Driver"
+	default y
+	depends on DT_HAS_REALTEK_RTL8211F_ENABLED
+	depends on MDIO
+	help
+	  Enable Realtek RTL8211F Ethernet PHY Driver
+
+if PHY_REALTEK_RTL8211F
+
+config PHY_REALTEK_RTL8211F_GPIO_SUPPORT
```

Comment:
I don't think you need a `Kconfig` for this, you can check using devicetree if a compatible has a `rst-gpios` property.

Reply:
That's how it was before in `c` code, but I've changed this after @JiafeiPan asked me to remove the mandatory dependency on `GPIO` (build problems on other platform). The dependency is added via `Kconfig` and it looks like the preprocessor currently does not have a function to tell if any compatible node instance has the property.

Reply:
The reason the kconfig is needed is to conditionalize the dependency on GPIO to avoid build errors, the PR was already as you're describing originally and that didn't work

Reply:
The macro has made it easier since #70385

Reply:
The PR used this macro as of a few days ago, the problem is not with the code file, the problem is with the KConfig dependency on GPIO, if it is unconditional then platforms without GPIO cannot build this driver

Reply:
Platforms without GPIO wouldn't add the gpio properties to the device instance?

Reply:
Yes, but how do you remove `depends on GPIO` from `Kconfig`?
That dependency would remove the driver sources from build.

Reply:
Where you do this now:
```c
#if CONFIG_PHY_REALTEK_RTL8211F_GPIO_SUPPORT
	if (gpio_is_ready_dt(&config->reset_gpio)) {
		// ....
	}
#endif /* CONFIG_PHY_REALTEK_RTL8211F_GPIO_SUPPORT */
```

do

```c
#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(rt_reset_gpio)
	if (gpio_is_ready_dt(&config->reset_gpio)) {
		// ....
	}
#endif /* DT_ANY_INST_HAS_PROP_STATUS_OKAY(rt_reset_gpio) */
```

Reply:
Again, this is what the PR was a few days ago. We changed it for a reason

Reply:
And I still fail to see the difficulty, maybe a more complete solution would be:

```c
#if defined(CONFIG_GPIO) && DT_ANY_INST_HAS_PROP_STATUS_OKAY(rt_reset_gpio)
#define RST_GPIO_SUPPORT 1
#endif

#if RST_GPIO_SUPPORT
// do something with gpio
#endif /* RST_GPIO_SUPPORT */
```

And you can `imply GPIO` in your Kconfig if you would like to default enable it.

Reply:
> And I still fail to see the difficulty, maybe a more complete solution would be:
> 
> ```c
> #if defined(CONFIG_GPIO) && DT_ANY_INST_HAS_PROP_STATUS_OKAY(rt_reset_gpio)
> #define RST_GPIO_SUPPORT 1
> #endif
> 
> #if RST_GPIO_SUPPORT
> // do something with gpio
> #endif /* RST_GPIO_SUPPORT */
> ```

this isn't the issue, the issue is the `depends on GPIO` is needed sometimes and not others, and I don't think we want to reverse the dependency here, @JiafeiPan can explain better maybe

Reply:
We cannot use `DT_ANY_INST_HAS_PROP_STATUS_OKAY` in `Kconfig` file to enable or disable `depends on GPIO`.
That's why there is the new redundant config item, to be able to bypass dependency on `GPIO` if it is not available on the platform.

Reply:
I won't block for this :) only a detail anyway.